### PR TITLE
New Redux Exercises

### PR DESCRIPTION
**Pull request by mdegani**
connect to #524 

- Removed old redux exercise
- Added new exercises

### CODE REVIEW

File: exercises/redux/1.0-SelectTodos_solution/app/app.component.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      <input type="checkbox" [checked]="todo.completed">
+      <label>{{todo.title}}</label>
+    </li>
+  </ul>
+  `
+})
+export class AppComponent {
+  @select() todos;
+  constructor(private ngRedux: NgRedux<any>) {
+    this.ngRedux.configureStore(todoApp, {}, []);
```
Comment (SethDavenport) [inline exercises/redux/1.0-SelectTodos_solution/app/app.component.ts]: If you're not using middleware you can probably omit the third argument.

Comment (mdegani) [inline exercises/redux/1.0-SelectTodos_solution/app/app.component.ts]: Thanks Seth.  I've made the change.

File: exercises/redux/1.0-SelectTodos_solution/app/app.component.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  <ul>
+    <li *ngFor="let todo of todos | async">
+      <input type="checkbox" [checked]="todo.completed">
+      <label>{{todo.title}}</label>
+    </li>
+  </ul>
+  `
+})
+export class AppComponent {
+  @select() todos;
```
Comment (e-schultz) [inline exercises/redux/1.0-SelectTodos_solution/app/app.component.ts]: `@select() todos$` - been having the naming convention of adding a `$` suffix to observables

File: exercises/redux/1.0-SelectTodos_solution/app/app.component.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { select } from 'ng2-redux';
+import  todoApp from './reducers/index';
+
+@Component({
+  selector: 'my-app',
+  styleUrls: ['app/app.component.css'],
+  template: `
+  <h2>todos</h2>
+  <ul>
+    <li *ngFor="let todo of todos | async">
```
Comment (e-schultz) [inline exercises/redux/1.0-SelectTodos_solution/app/app.component.ts]:
```
<li *ngFor="let todo of todos$ | async">
```

File: exercises/redux/2.0-DispatchActions/app/app.component.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  <ul>
+    <li *ngFor="let todo of todos | async">
+      <input type="checkbox" [checked]="todo.completed">
+      <label>{{todo.title}}</label>
+    </li>
+  </ul>
+  `
+})
+export class AppComponent {
+  @select() todos;
```
Comment (e-schultz) [inline exercises/redux/2.0-DispatchActions/app/app.component.ts]:
```
 @select() todos$
```

File: exercises/redux/2.0-DispatchActions/app/app.component.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { select } from 'ng2-redux';
+import  todoApp from './reducers/index';
+
+@Component({
+  selector: 'my-app',
+  styleUrls: ['app/app.component.css'],
+  template: `
+  <h2>todos</h2>
+  <ul>
+    <li *ngFor="let todo of todos | async">
```
Comment (e-schultz) [inline exercises/redux/2.0-DispatchActions/app/app.component.ts]:
```
<li *ngFor="let todo of todos$ | async">
```

File: exercises/redux/2.0-DispatchActions_solution/app/app.component.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import  todoApp from './reducers/index';
+import * as TodoActions from './actions/index';
+
+@Component({
+  selector: 'my-app',
+  styleUrls: ['app/app.component.css'],
+  template: `
+  <h2>todos</h2>
+  <ul>
+    <li *ngFor="let todo of todos | async">
```
Comment (e-schultz) [inline exercises/redux/2.0-DispatchActions_solution/app/app.component.ts]:
```
<li *ngFor="let todo of todos$ | async">
```

File: exercises/redux/2.0-DispatchActions_solution/app/app.component.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      <input type="checkbox" [checked]="todo.completed" (change)="toggleTodo(todo.id)">
+      <label [style.text-decoration]="todo.completed ? 'line-through' : 'none'">{{todo.title}}</label>
+    </li>
+  </ul>
+  `
+})
+export class AppComponent {
+  @select() todos;
```
Comment (e-schultz) [inline exercises/redux/2.0-DispatchActions_solution/app/app.component.ts]:
```
@select() todos$;
```

**Comment by e-schultz** [comment]
@mdegani some minor tweaks for naming conventions of observables, but otherwise 👍

File: exercises/redux/1.0-SelectTodos_solution/app/app.component.ts (hunk 2 above)
Comment (mdegani) [inline exercises/redux/1.0-SelectTodos_solution/app/app.component.ts]: Thanks Evan.  I've made the changes.  *Now* I get why ng2-redux ignores `$`'s 😄

**Pull request merged**